test/boost/sstable_compaction_test.cc

```diff
@@ -2311,93 +2311,80 @@ SEASTAR_TEST_CASE(sstable_validate_test) {
-    return do_with_cql_env([] (cql_test_env& cql_env) -> future<> {
-        return test_env::do_with_async([] (test_env& env) {
-            auto schema = schema_builder("ks", get_name())
-                    .with_column("pk", utf8_type, column_kind::partition_key)
-                    .with_column("ck", int32_type, column_kind::clustering_key)
-                    .with_column("s", int32_type, column_kind::static_column)
-                    .with_column("v", int32_type).build();
-            auto permit = env.make_reader_permit();
-
-            std::vector<mutation_fragment_v2> scrubbed_fragments;
-            auto sst = env.make_sstable(schema);
-
-            const auto corrupt_fragments = write_corrupt_sstable(env, *sst, permit, [&] (mutation_fragment_v2&& mf, bool is_corrupt) {
-                if (!is_corrupt) {
-                    scrubbed_fragments.emplace_back(std::move(mf));
-                }
-            });
-
-            testlog.info("Writing sstable {}", sst->get_filename());
-
-            sst->load(sst->get_schema()->get_sharder()).get();
-
-            testlog.info("Loaded sstable {}", sst->get_filename());
+SEASTAR_THREAD_TEST_CASE(sstable_scrub_abort_mode_test) {
+    scrub_test_framework test;
 
-            auto table = env.make_table_for_tests(schema);
-            auto close_cf = deferred_stop(table);
-            table->start();
-            auto& compaction_manager = table->get_compaction_manager();
+    auto schema = test.schema();
 
-            table->add_sstable_and_update_cache(sst).get();
+    auto muts = tests::generate_random_mutations(test.random_schema(), 3).get();
+    std::swap(*muts.begin(), *(muts.begin() + 1));
 
-            bool found_sstable = false;
-            foreach_table_state_with_thread(table, [&] (compaction::table_state& ts) {
-                auto sstables = in_strategy_sstables(ts);
-                if (sstables.empty()) {
-                    return;
-                }
+    test.run(schema, muts, [] (table_for_tests& table, compaction::table_state& ts, std::vector<sstables::shared_sstable> sstables) {
                 BOOST_REQUIRE(sstables.size() == 1);
-                BOOST_REQUIRE(sstables.front() == sst);
-                found_sstable = true;
-
-                auto verify_fragments = [&] (sstables::shared_sstable sst, const std::vector<mutation_fragment_v2>& mfs) {
-                    auto r = assert_that(sst->as_mutation_source().make_reader_v2(schema, permit));
-                    for (const auto& mf : mfs) {
-                       testlog.trace("Expecting {}", mutation_fragment_v2::printer(*schema, mf));
-                       r.produces(*schema, mf);
-                    }
-                    r.produces_end_of_stream();
-                };
-
-                testlog.info("Verifying written data...");
-
-                // Make sure we wrote what we though we wrote.
-                verify_fragments(sst, corrupt_fragments);
+                auto sst = sstables.front();
```

Comment:
A thought: Do you actually have a case where the test runner is supposed to return more than one sstable?
If there isn't, you can mode the `assert(sstables.size()==1); sst=sstables.front();` code into the test framework and it will only return a single sstable, always.

Reply:
Yes there are, depending on what you pass as parameters to the test framework.
I'm a bit fuzzy on the details now (I wrote this 2 months ago), but I remember planning to write tests which would have more than one sstables, so I didn't encode this into the framework.